src/utilities/routerNavigation.ts

```diff
@@ -61,42 +62,28 @@ function createBrowserNavigation({ onBeforeLocationUpdate, onAfterLocationUpdate
       }
     }
 
-    await updateBrowserUrl(url, options)
+    action?.()
```

Comment:
should this be awaited? we will need to use `if(action)` rather than the safe navigation if we're going to await it

Reply:
its not async so I don't think there's a reason to await it